### PR TITLE
refactor: remove explore from visualization provider

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #8153

### Description:

Main changes:
- remove explore from VisualizationProvider props
- remove explore, metrics, custom metrics, table calculation, explore name from the VisualizationProvider context
- refactor some stuff that was missed in the previous PRs
  - in some cases we just need to remove the explore code
  - in some cases we need to replace the explore with itemsMap

Tested: explore, saved chart, SQL runner, minimal dashboard, minimal chart

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by rephus** [comment]
Date zoom seems broken here https://lightdash-pr-8239.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/dashboards/a982e2a5-1acf-44b8-8026-52cb9893f63d?dateZoom=day
[Screencast from 30-11-23 11:46:32.webm](https://github.com/lightdash/lightdash/assets/1983672/d7b5edfd-05fd-4592-ae20-e68b6fe3e43f)

**Comment by joaoviana** [comment]
> Reference lines are a bit funny
> 
> * I can't add values to dates
> * Adding a value to a number metric, crops the chart.
> 
>  Screencast.from.08-12-23.13.41.25.webm 
> I'm not 100% sure this is caused by you changes, but it might be possible

on 
> I can't add values to dates

that is correct, https://github.com/lightdash/lightdash/blob/refactor/remove-explore-from-visualization-provider/packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/Legend/ReferenceLine.tsx#L151 - here we disable if the `item` is non-numeric

on
> Adding a value to a number metric, crops the chart.

on `main` the same happens as well: 
image

🤔

File: packages/backend/src/database/seeds/development/02_saved_queries.ts [hunk 1]
```diff
@@ -145,29 +145,29 @@ export async function seed(knex: Knex): Promise<void> {
...
+                        fieldId: 'orders_order_date_day',
                         descending: false,
                     },
                 ],
                 tableCalculations: [
                     {
                         name: 'cumulative_order_count',
                         displayName: 'Cumulative order count',
-                        sql: 'SUM(${orders.unique_order_count})\nOVER(ORDER BY ${orders.order_date})',
+                        sql: 'SUM(${orders.unique_order_count})\nOVER(ORDER BY ${orders.order_date_day})',
```
Comment (joaoviana) [inline packages/backend/src/database/seeds/development/02_saved_queries.ts:162]: which is why for that chart it wasn't working @rephus ;

**Comment by joaoviana** [comment]
> Date zoom seems broken here https://lightdash-pr-8239.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/dashboards/a982e2a5-1acf-44b8-8026-52cb9893f63d?dateZoom=day
> 
>  Screencast.from.08-12-23.13.35.40.webm

going to replicate this!

**Comment by joaoviana** [comment]
> Date zoom seems broken here https://lightdash-pr-8239.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/dashboards/a982e2a5-1acf-44b8-8026-52cb9893f63d?dateZoom=day
> 
>  Screencast.from.08-12-23.13.35.40.webm

and this is fixed as well!

https://github.com/lightdash/lightdash/assets/7611706/e26f784c-beff-479b-b1f9-8ce545dfb45e

**Review by joaoviana — APPROVED** [review]

**Pull request merged**